internal/host/hostcommands/install_cmd.go

```diff
@@ -181,6 +182,22 @@ func (i *installCmd) getFullInstallerCommand(cluster *common.Cluster, host *mode
 		shellescape.QuoteCommand(installerCmd)), nil
 }
 
+func (i *installCmd) getMustGatherArgument(mustGatherMap versions.MustGatherVersion) string {
+	//for backward compatability, if must gather images map contains only the ocp must gather
+	//we shall send a single image. otherwise, we shall send a json structure holding all the
+	//relevant images
+	if len(mustGatherMap) == 1 && mustGatherMap["ocp"] != "" {
+		return mustGatherMap["ocp"]
+	}
+
+	arg, err := json.Marshal(mustGatherMap)
+	if err != nil {
+		i.log.WithError(err).Errorf("can not encode must-gather image map")
+		return ""
```

Comment:
Are we ok with not exposing the error? I mean, it indeed looks clean returning only the argument, but we usually return the err as well. In this case, so we could early return in 'getFullInstallerCommand', and won't invoke the cmd with an empty must gather image.

Reply:
changes